### PR TITLE
helm: update to latest ibm-skydive-dev chart

### PR DESCRIPTION
**Pull request by hunchback**
aligned our chart content with https://github.com/IBM/charts/tree/master/stable/ibm-skydive-dev

marked as WIP until completing testing

### CODE REVIEW

**Comment by hunchback** [comment]
I tested deployment:

```
scripts/ci/install-helm.sh install
scripts/ci/install-helm.sh start
```

the installing the helm chart:

```
cd contrib/charts
[vagrant@dev charts]$ helm delete --purge  my-release
release "my-release" deleted
[vagrant@dev charts]$ helm install --name my-release .
NAME:   my-release
LAST DEPLOYED: Tue Aug 27 14:19:42 2019
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1/Pod(related)
NAME                                         READY  STATUS             RESTARTS  AGE
my-release-skydive-agent-sxmzn               0/1    ContainerCreating  0         1s
my-release-skydive-analyzer-97557769f-xkfx4  0/1    ContainerCreating  0         1s

==> v1/ServiceAccount
NAME                     SECRETS  AGE
skydive-service-account  1        1s

==> v1/ClusterRoleBinding
NAME                          AGE
skydive-cluster-role-binding  1s

==> v1/Service
NAME                                 TYPE       CLUSTER-IP      EXTERNAL-IP  PORT(S)             AGE
my-release-skydive-internal-service  ClusterIP  10.107.141.222  <none>       12379/TCP,9200/TCP  1s
my-release-skydive-service           NodePort   10.106.136.255  <none>       8082:30504/TCP      1s

==> v1beta1/DaemonSet
NAME                      DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR  AGE
my-release-skydive-agent  1        1        0      1           0          <none>         1s

==> v1beta1/Deployment
NAME                         DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
my-release-skydive-analyzer  1        1        1           0          1s


NOTES:
To access skydive GUI execute:

(Bear with us couple of seconds until Skydive starts ...)
export UI_PORT=$(kubectl get --namespace default -o jsonpath="{.spec.ports[0].nodePort}" services my-release-skydive-service)
export UI_IP=$(kubectl get nodes --namespace default -o jsonpath="{.items[0].status.addresses[0].address}")
echo "skydive end-point: http://$UI_IP:$UI_PORT"



[vagrant@dev charts]$ (Bear with us couple of seconds until Skydive starts ...)
-bash: Bear: command not found
[vagrant@dev charts]$ export UI_PORT=$(kubectl get --namespace default -o jsonpath="{.spec.ports[0].nodePort}" services my-release-skydive-service)
[vagrant@dev charts]$ export UI_IP=$(kubectl get nodes --namespace default -o jsonpath="{.items[0].status.addresses[0].address}")
[vagrant@dev charts]$ echo "skydive end-point: http://$UI_IP:$UI_PORT"
skydive end-point: http://10.0.2.15:30504
```

**Comment by hunchback** [comment]
got the following view when running the chart on minikube: 
image

**Comment by hunchback** [comment]
> Can you add back the updated values-metadata file.

I intentionally removed any non-standard parts from the helm (such as values-metadata and pak) with the intent that we can contribute this Helm to https://github.com/helm/charts

**Comment by hunchback** [comment]
run skydive-cdd-overview-tests

**Comment by hunchback** [comment]
run skydive-helm-tests

**Comment by hunchback** [comment]
run skydive-helm-tests

**Comment by hunchback** [comment]
run skydive-cdd-overview-tests

**Comment by hunchback** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by hunchback** [comment]
run skydive-cdd-overview-tests

File: contrib/charts/templates/deployment-analyzer.yaml [hunk 1]
```diff
@@ -17,56 +17,110 @@ spec:
...
         app: {{ template "fullname" . }}
+        chart: "{{ .Chart.Name }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
         tier: analyzer
     spec:
+      serviceAccountName: skydive-service-account
+      hostNetwork: true
+      dnsPolicy: ClusterFirstWithHostNet
+      hostPID: true
```
Comment (lebauce) [inline contrib/charts/templates/deployment-analyzer.yaml]: Not sure this is required

File: contrib/charts/templates/deployment-analyzer.yaml [hunk 2]
```diff
@@ -17,56 +17,110 @@ spec:
         productVersion: {{ .Chart.Version }}
       labels:
         app: {{ template "fullname" . }}
+        chart: "{{ .Chart.Name }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
         tier: analyzer
     spec:
+      serviceAccountName: skydive-service-account
+      hostNetwork: true
```
Comment (lebauce) [inline contrib/charts/templates/deployment-analyzer.yaml]: This is not required for the analyzer

File: contrib/charts/templates/deployment-analyzer.yaml [hunk 3]
```diff
@@ -17,56 +17,110 @@ spec:
...
+        - containerPort: {{ .Values.etcd.port }}
+        securityContext:
+          privileged: false
+          readOnlyRootFilesystem: false
+          allowPrivilegeEscalation: false
+          runAsNonRoot: false
+          runAsUser: 0
+          capabilities:
+                add:
+                - SYS_ADMIN
```
Comment (lebauce) [inline contrib/charts/templates/deployment-analyzer.yaml]: Why giving these previleges to the analyzer ?

File: contrib/charts/templates/deployment-analyzer.yaml (hunk 1 above)
Comment (hunchback) [reply contrib/charts/templates/deployment-analyzer.yaml]: removed

File: contrib/charts/templates/deployment-analyzer.yaml (hunk 2 above)
Comment (hunchback) [reply contrib/charts/templates/deployment-analyzer.yaml]: ok

File: contrib/charts/templates/deployment-analyzer.yaml (hunk 3 above)
Comment (hunchback) [reply contrib/charts/templates/deployment-analyzer.yaml]: removed

**Comment by hunchback** [comment]
run skydive-helm-tests

**Comment by hunchback** [comment]
run skydive-functional-hw-tests
run skydive-functional-tests-backend-elasticsearch
run skydive-cdd-overview-tests

**Comment by hunchback** [comment]
run skydive-functional-hw-tests
run skydive-cdd-overview-tests

**Comment by hunchback** [comment]
run skydive-functional-hw-tests

**Pull request merged**